extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -757,6 +766,28 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     return this.getAdUrlDeferred.promise;
   }
 
+  /** @override */
+  getServeNpaSignal() {
+    const npaSignal = this.element.getAttribute('always-serve-npa');
+    if (npaSignal == undefined) {
+      return Promise.resolve(false);
+    }
+    if (npaSignal == '') {
+      return Promise.resolve(true);
+    }
+    return Services.geoForDocOrNull(this.element).then((geoService) => {
+      if (geoService) {
+        const locations = npaSignal.split(',');
+        for (let i = 0; i < locations.length; i++) {
+          if (geoService.isInCountryGroup(locations[i]) === GEO_IN_GROUP.IN) {
+            return true;
+          }
+        }
+      }
+      return false;
+    });
+  }
+
```

Comment:
To avoid having to duplicate this code, could it go in ads/google/a4a/utils.js or somewhere else shared?

Reply:
Thanks for this -- didn't know it `utils.js` existed